packages/react/src/ReactElementValidator.js

```diff
@@ -333,6 +333,10 @@ export function createElementWithValidation(type, props, children) {
       typeString = 'array';
     } else {
       typeString = typeof type;
+      if (type && type.hasOwnProperty('$$typeof')) {
+        typeString = 'element';
```

Comment:
"React element" would be better here.

Also now that I think of it let's compare `type.$$typeof === REACT_ELEMENT_TYPE` explicitly. We don't actually "own" this convention. You can get `REACT_ELEMENT_TYPE` from existing `ReactSymbols` import in this file.